compiler/optimizer/VirtualGuardHeadMerger.cpp

```diff
@@ -342,8 +342,8 @@ int32_t TR_VirtualGuardHeadMerger::perform() {
             // tail split
             if ((block->getNextBlock()->getSuccessors().size() == 1) ||
                 ((block->getNextBlock()->getSuccessors().size() == 2) &&
-                 block->getNextBlock()->getLastRealTreeTop()->getNode()->getOpCode().isBranch()) &&
-                performTransformation(comp(), "%sCloning block_%d and placing clone after block_%d to reduce HCR guard nops\n", OPT_DETAILS, block->getNextBlock()->getNumber(), cold1->getNumber()))
+                 block->getNextBlock()->getLastRealTreeTop()->getNode()->getOpCode().isBranch() &&
```

Comment:
I'm going a little cross-eyed trying to read this, but I think the intention is for the performTransformation to be applied to both size == 1 and size == 2 which is not what the change does - it just bind the perform to the size == 2 case

Reply:
yeah...looking at it again, I think you're right. I'll fix it up.